### ISSUE TITLE
Close console in input(), "Spyder has encountered an internal problem!"

### ISSUE THREAD

**Issue by nerdfever** [spyder-ide/spyder#15207]
### What steps will reproduce the problem?

Spyder 5.0.0, Win10 x64

Click "run" button on a program that includes input().

When program is stopped waiting for input, close Python instance in IPython pane (click X button on tab).

You immediately get "Spyder has encountered an internal problem!".

You can close the Issue Reporter box and continue if you want.

### Traceback
```python-traceback
--- Logging error ---
Traceback (most recent call last):
  File "C:\Program_Files\anaconda3\lib\site-packages\qtconsole\comms.py", line 113, in get_comm
    return self.comms[comm_id]
KeyError: '09a25292994f11ebb9a4b42e99c1a286'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Program_Files\anaconda3\lib\logging\__init__.py", line 1084, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "C:\Program_Files\anaconda3\Scripts\spyder-script.py", line 10, in <module>
    sys.exit(main())
  File "C:\Program_Files\anaconda3\lib\site-packages\spyder\app\start.py", line 214, in main
    mainwindow.main(options, args)
  File "C:\Program_Files\anaconda3\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "C:\Program_Files\anaconda3\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "C:\Program_Files\anaconda3\lib\site-packages\qtconsole\comms.py", line 48, in _dispatch
    getattr(self, msg_type)(msg)
  File "C:\Program_Files\anaconda3\lib\site-packages\qtconsole\comms.py", line 172, in comm_msg
    comm = self.get_comm(comm_id)
  File "C:\Program_Files\anaconda3\lib\site-packages\qtconsole\comms.py", line 117, in get_comm
    self.log.warning("No such comm: %s", comm_id)
Message: 'No such comm: %s'
Arguments: ('09a25292994f11ebb9a4b42e99c1a286',)
```

## Versions

* Spyder version: 5.0.0 
* Python version: 3.8.5
* Qt version: 5.9.7
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  1.6.0 (OK)
cookiecutter >=1.6.0          :  1.7.2 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.6.0               :  7.22.0 (OK)
jedi =0.17.2                  :  0.17.2 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  22.3.0 (OK)
nbconvert >=4.0               :  6.0.7 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
paramiko >=2.4.0              :  2.7.2 (OK)
parso =0.7.0                  :  0.7.0 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.8.0 (OK)
pygments >=2.0                :  2.8.1 (OK)
pylint >=1.0                  :  2.7.4 (OK)
pyls >=0.36.2;<1.0.0          :  0.36.2 (OK)
pyls_black >=0.4.6            :  0.4.6 (OK)
pyls_spyder >=0.3.2           :  0.3.2 (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.1.10 (OK)
qtawesome >=0.5.7             :  1.0.2 (OK)
qtconsole >=5.0.3             :  5.0.3 (OK)
qtpy >=1.5.0                  :  1.9.0 (OK)
rtree >=0.8.3                 :  0.9.4 (OK)
setuptools >=39.0.0           :  52.0.0.post20210125 (OK)
sphinx >=0.6.6                :  3.5.3 (OK)
spyder_kernels >=2.0.1,<2.1.0 :  2.0.1 (NOK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3;<2.0.0      :  1.0.2 (OK)
zmq >=17                      :  20.0.0 (OK)

# Optional:
cython >=0.21                 :  0.29.22 (OK)
matplotlib >=2.0.0            :  3.3.4 (OK)
numpy >=1.7                   :  1.19.2 (OK)
pandas >=1.1.1                :  1.2.3 (OK)
scipy >=0.17.0                :  1.6.2 (OK)
sympy >=0.7.3                 :  1.7.1 (OK)
```

**Comment by ccordoba12** [comment]
Hey @nerdfever, thanks for reporting. I couldn't reproduce this on Linux.

@dalthviz, could you try it on Windows?

**Comment by dalthviz** [comment]
@ccordoba12 I was unable to reproduce this.

@nerdfever could you share with us an example code that triggers this error consistently for you? (Maybe the code to be run needs more than an `input` call to trigger the error) Thanks!

**Comment by nerdfever** [comment]
It's strange - I've been getting that crash dozens of times all day - every time I close the Python interpreter when a program is running.

I found it would do it with this:

```
def main():
    for i in range(int(1e12)):
        pass

if __name__ == '__main__':
    main()
```

Then I went to record it as a video, and it's stopped doing it. Of course.

I'm sorry - problem is probably located between users' ears. If I find a way to reproduce it I'll let you know.
